HeterogeneousCore/CUDACore/python/ProcessAcceleratorCUDA.py

```diff
@@ -4,27 +4,40 @@
 
 class ProcessAcceleratorCUDA(cms.ProcessAccelerator):
     def __init__(self):
-        super(ProcessAcceleratorCUDA,self).__init__()
+        super(ProcessAcceleratorCUDA, self).__init__()
         self._label = "gpu-nvidia"
+        self._required_tool = "cuda-gcc-support"
+        self._runtime_check = "cudaIsEnabled"
+        self._available = (os.system(f"scram tool tag {self._required_tool} TOOLNAME >& /dev/null") == 0)
```

Comment:
I would strongly prefer to not make the configuration system depend on our build system. (also test failures look like this doesn't work properly in the PR tests?)

A simple replacement could be to change https://github.com/cms-sw/cmssw/blob/f459c014c95f3fc69bbc36da83c103e57789b870/HeterogeneousCore/CUDAUtilities/bin/cudaIsEnabled_fallback.cpp#L5
to e.g.
```cpp
int main() { return 2; }
```
and use the return value to check if the CUDA/ROCm is not available in the build (although I'm still not convinced if the disambiguation of that from "not being able to run on CUDA/ROCm" is useful in this place).

Reply:
> (also test failures look like this doesn't work properly in the PR tests?)

I don't know... it's the whole week that I get errors and test failures that I am unable to reproduce locally, so I'm kind of stumped.